### PR TITLE
Fix infinite parsing loop caused by stray semicolon

### LINKED ISSUE #6916: Stray semi-colon cause infinite loop in IdfParser
Issue overview
--------------
User file has a stray semi-colon by itself between complete objects. This results in an infinite loop [here](https://github.com/NREL/EnergyPlus/blob/d79a0f2f2cd90a2d57a44b3b3ceefbc05d87a0f3/src/EnergyPlus/InputProcessing/IdfParser.cc#L202-L220) in IdfParse that just keeps chugging away eating more and more memory as it goes. Reproduced with a smaller example file.

[1ZoneUncontrolled-StraySemiColon.idf.txt](https://github.com/NREL/EnergyPlus/files/2300686/1ZoneUncontrolled-StraySemiColon.idf.txt)

### Details
Some additional details for this issue (if relevant):
 - Win64
 - Version of EnergyPlus Version 9.0.0-d79a0f2f2c
 - Sort of related to helpdesk ticket 11958

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mbadams5**
Pull request overview
---------------------
Fixes #6916

This fixes an infinite parsing loop caused by a stray semicolon in the input file. It will also now catch stray comma's as well, throwing an error. The stray semicolon does not throw an error and instead just continues parsing.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,7 +35,7 @@ Build
 Builds
 
 # by default CLion uses cmake-build-debug as the build folder, so add that as well
-cmake-build-debug
+cmake-build-*
```
Comment (Myoldmopar) [inline .gitignore:38]: :+1:

File: tst/EnergyPlus/unit/InputProcessor.unit.cc [hunk 2]
```diff
@@ -261,8 +261,7 @@ TEST_F(InputProcessorFixture, decode_encode_2)
                                           "  Counterclockwise,",
                                           "  Relative,",
                                           "  Relative,",
-                                          "  Relative;"
-                                          "",
+                                          "  Relative;",
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/InputProcessor.unit.cc:264]: This unit test doesn't seem to cover the proposed fix here.  I'll modify a test file to verify the fix, but should we also create a unit test for it here?

Comment (mbadams5) [reply tst/EnergyPlus/unit/InputProcessor.unit.cc:264]: It doesn't. I just cleaned up this unit test.

I tried to think of a way to unit test this but wasn't successful in my fairly quick attempt to fix this. The problem is that google test doesn't seem to have timeout functionality so how do you test for something that causes infinite looping?

There is a defect file for this issue though.

**Comment by Myoldmopar** [comment]
Ahh, that makes good sense @mbadams5.  I went ahead and tested locally.

#### Develop Branch
 - An unmodified 1ZoneUncontrolled.idf file of course runs fine.  
 - I modified it with a stray comma, and it gives `** Severe  ** Line: 92 Index: 0 - "" is not a valid Object Type.`
 - I modified it with a stray semicolon instead, and it runs for 65 seconds on my machine, essentially stuck in an infinite loop, until finally failing.  I'm guessing it's hitting a stack overflow if it is a recursive problem.  Anyway, it eventually aborted without a helpful message.

#### This Branch
 - Unmodified file passes fine
 - Stray comma file fails with ` ** Severe  ** Line: 91 Index: 13 - Extraneous comma found.`
 - Stray semicolon file passes without issue.

This is good to go.  Merging.

**Pull request merged**